lib/app/services/audio_wave_playback_service/audio_wave_playback_service.dart

```diff
@@ -60,9 +62,11 @@ class AudioWavePlaybackService {
       name: id,
       link: LinkDetails(link: audioUrl),
     );
-    return savedFilePath;
+    final compressedFilePath = await _mediaCompressionService.compressAudioToWav(savedFilePath);
+    return compressedFilePath;
   }
 }
 
 @riverpod
-AudioWavePlaybackService audioWavePlaybackService(Ref ref) => const AudioWavePlaybackService();
+AudioWavePlaybackService audioWavePlaybackService(Ref ref) =>
+    AudioWavePlaybackService(ref.read(mediaCompressServiceProvider));
```

Comment:
Shouldn't it be `ref.watch`?

Reply:
why do you think we should use `watch`?

Reply:
It's a recommended way to use `ref.watch `in this type of provider and according to Riverpod's documentation - https://riverpod.dev/docs/essentials/combining_requests#the-refwatch-method